src/main/java/taskmaster/parser/Parser.java

```diff
@@ -0,0 +1,131 @@
+     *
+     * @param userInput String input by the user
+     * @throws DeadlineCommandMissingInputException    If deadline command missing input for task and/or by field
+     * @throws DeadlineCommandWrongSubCommandException If deadline command does not have /by sub command
+     * @throws DeleteCommandMissingInputException      If delete command missing input for index field
+     * @throws DeleteCommandOutOfBoundsException       If delete command used with non-existing index
+     * @throws DeleteCommandTooManyInputException      If delete command has spaces in index field
+     * @throws EmptyTodoTaskException                  If todo command missing input for task field
+     * @throws EventCommandMissingInputException       If event command missing input for task and/or from and/or to field
+     * @throws EventCommandWrongSubCommandException    If event command does not have /from and/or /to command
+     * @throws FindCommandMissingInputException        If find command missing input for keyword field
+     * @throws MarkCommandMissingInputException        If mark command missing input for index field
+     * @throws MarkCommandTooManyInputException        If mark command has spaces in index field
+     * @throws MarkUnmarkOutOfBoundsException          If mark or unmark command used with non-existing index
+     * @throws UnmarkCommandMissingInputException      If unmark command missing input for index field
+     * @throws UnmarkCommandTooManyInputException      If unmark command has spaces in index field
+     */
+    private void handleCommand(String userInput)
+            throws DeadlineCommandMissingInputException, DeadlineCommandWrongSubCommandException,
+            DeleteCommandMissingInputException, DeleteCommandOutOfBoundsException,
+            DeleteCommandTooManyInputException, EmptyTodoTaskException,
+            EventCommandMissingInputException, EventCommandWrongSubCommandException,
+            FindCommandMissingInputException, MarkCommandMissingInputException,
+            MarkCommandTooManyInputException, MarkUnmarkOutOfBoundsException,
+            UnmarkCommandMissingInputException, UnmarkCommandTooManyInputException {
+        if (userInput.startsWith("list")) {
+            taskList.listTasks();
+
+        } else if (userInput.startsWith("mark")) {
+            taskList.markTask(userInput);
+
+        } else if (userInput.startsWith("unmark")) {
+            taskList.unmarkTask(userInput);
+
+        } else if (userInput.startsWith("todo")) {
+            taskList.addToDo(userInput);
+
+        } else if (userInput.startsWith("deadline")) {
+            taskList.addDeadline(userInput);
+
+        } else if (userInput.startsWith("event")) {
+            taskList.addEvent(userInput);
+
+        } else if (userInput.startsWith("delete")) {
+            taskList.deleteTask(userInput);
+
+        } else if (userInput.startsWith("find")) {
+            taskList.findTask(userInput);
+
+        } else {
```

Comment:
consider uding constants for the command words
for better readability, you may consider using switch statement.